packages/block-library/src/template-part/variations.js

```diff
@@ -6,55 +6,57 @@ import { store as editorStore } from '@wordpress/editor';
 import { store as blocksStore } from '@wordpress/blocks';
 import { dispatch, select, subscribe } from '@wordpress/data';
 
-const unsubscribe = subscribe( () => {
-	const definedVariations = select(
-		editorStore
-	).__experimentalGetDefaultTemplatePartAreas();
+export default function subscribeToCreateVatiations() {
+	const unsubscribe = subscribe( () => {
+		const definedVariations = select(
+			editorStore
+		).__experimentalGetDefaultTemplatePartAreas();
```

Comment:
I'm not sure if it'd be useful in this case, but just wanted to note it's possible to register variations via `register_block_type_from_metadata` eg

https://github.com/WordPress/gutenberg/blob/ff4c3c4094c730be0ea808ce868ebc220ad2320f/packages/block-library/src/navigation-link/index.php#L325-L331

There is still some JS needed for icon setting/isActive logic, since there's a bit more API work to be done to help serialize that.

Reply:
Ah I see you're already created https://github.com/WordPress/gutenberg/pull/31761 👍

Reply:
Yes, thank you!  I saw your PR registering them that way and it works great here. 😁

Reply:
oddly that approach doesn't seem to work with everyones testing.  For some reason a couple folks seem to have that filter run before the actual php side registers the variations (?)- https://github.com/WordPress/gutenberg/pull/31761#issuecomment-840516209

Reply:
See example in https://github.com/WordPress/gutenberg/pull/29095

Server variation support landed in WP 5.8, so there's still a need for fallback variations with WP 5.7 and below, until Gutenberg requires at least 5.8.

https://github.com/WordPress/gutenberg/blob/815b767a46f03513143b15cbd200f213306f7d18/gutenberg.php#L6

Reply:
That makes sense, thanks!